preprocess/preprocess.wasm.go

```diff
@@ -0,0 +1,220 @@
+//go:build wasm
```

Comment:
What is the basic difference in the code between preprocess.wasm.go and preprocess.go

Reply:
there is a lot of code repetition there converted it into 3 separate file to remove duplication

Reply:
The difference is in function calls , wasm has 1 parameter , cli has 2 params , each of them calls different functions